### ISSUE TITLE
Fatal error when starting workspace

### ISSUE THREAD

**Issue by AndreGKruger** [loft-sh/devpod#805]
**What happened?**  
Starting a new workspace fails with a fatal error

**What did you expect to happen instead?**  
Workspace should start

**How can we reproduce the bug?** (as minimally and precisely as possible)  

My *`devcontainer.json`*:
```
{
    "image":"mcr.microsoft.com/devcontainers/go",
    "features": {
		"ghcr.io/devcontainers/features/node:1":{}
	},
    "customizations": {
		"vscode": {
			"settings": {},
			"extensions": [
				"streetsidesoftware.code-spell-checker",
                "mitchdenny.ecdc",
                "GitHub.copilot",
                "GitHub.copilot-chat",
                "golang.go",
                "bradlc.vscode-tailwindcss",
                "lightyen.tailwindcss-intellisense-twin",
                "Postman.postman-for-vscode"
			]
		}
	},
    "forwardPorts": [8080],
}
```

**Local Environment:**  
- DevPod Version: 0.4.2
- Operating System: mac
- ARCH of the OS: ARM64 

**DevPod Provider:**  
- Cloud Provider: onprem
- Kubernetes Provider: v1.28.2
- Local/remote provider: ssh
- Custom provider: n/a

**Anything else we need to know?**  

**Logs**
```
[13:26:33] debug SSH client created
[13:26:33] debug SSH session created
[13:26:34] info Exit error: start agent: stat /tmp: no such file or directory
[13:26:34] debug Connection to SSH Server closed
[13:26:34] info start agent: stat /tmp: no such file or directory
[13:26:34] debug Done setting up container
[13:26:34] info Process exited with status 1
[13:26:34] info run agent command
[13:26:34] info github.com/loft-sh/devpod/pkg/devcontainer/sshtunnel.ExecuteCommand.func2
[13:26:34] info /Users/runner/work/devpod/devpod/pkg/devcontainer/sshtunnel/sshtunnel.go:111
[13:26:34] info runtime.goexit
[13:26:34] info /Users/runner/hostedtoolcache/go/1.20.5/x64/src/runtime/asm_arm64.s:1172
[13:26:34] info devcontainer up
[13:26:34] info github.com/loft-sh/devpod/cmd/agent/workspace.(*UpCmd).Run
[13:26:34] info /Users/runner/work/devpod/devpod/cmd/agent/workspace/up.go:95
[13:26:34] info github.com/loft-sh/devpod/cmd/agent/workspace.NewUpCmd.func1
[13:26:34] info /Users/runner/work/devpod/devpod/cmd/agent/workspace/up.go:52
[13:26:34] info github.com/spf13/cobra.(*Command).execute
[13:26:34] info /Users/runner/work/devpod/devpod/vendor/github.com/spf13/cobra/command.go:916
[13:26:34] info github.com/spf13/cobra.(*Command).ExecuteC
[13:26:34] info /Users/runner/work/devpod/devpod/vendor/github.com/spf13/cobra/command.go:1044
[13:26:34] info github.com/spf13/cobra.(*Command).Execute
[13:26:34] info /Users/runner/work/devpod/devpod/vendor/github.com/spf13/cobra/command.go:968
[13:26:34] info github.com/loft-sh/devpod/cmd.Execute
[13:26:34] info /Users/runner/work/devpod/devpod/cmd/root.go:90
[13:26:34] info main.main
[13:26:34] info /Users/runner/work/devpod/devpod/main.go:8
[13:26:34] info runtime.main
[13:26:34] info /Users/runner/hostedtoolcache/go/1.20.5/x64/src/runtime/proc.go:250
[13:26:34] info runtime.goexit
[13:26:34] info /Users/runner/hostedtoolcache/go/1.20.5/x64/src/runtime/asm_arm64.s:1172
[13:26:34] debug Connection to SSH Server closed
[13:26:34] debug Done creating devcontainer
[13:26:34] debug Done executing ssh server helper command
[13:26:34] fatal Process exited with status 1
run agent command
github.com/loft-sh/devpod/pkg/devcontainer/sshtunnel.ExecuteCommand.func2
        /Users/runner/work/devpod/devpod/pkg/devcontainer/sshtunnel/sshtunnel.go:111
runtime.goexit
        /Users/runner/hostedtoolcache/go/1.20.5/x64/src/runtime/asm_arm64.s:1172
```

**Comment by kfox1111** [comment]
I'm seeing the same thing when trying to run it against minikube.

**Comment by AndreGKruger** [comment]
Looks like the problem starts at : ```info Exit error: start agent: stat /tmp: no such file or directory```

**Comment by yufio** [comment]
I get the same issue trying to create my workspace with docker as a provider.

**Comment by yufio** [comment]
hi @FabianKramm, do you by chance already know when you plan the next release that includes this fix?

**Comment by kfox1111** [comment]
I just tested a checkout from main and it is working for me. Thanks!

**Comment by sanguivore-easyco** [comment]
@yufio I'm running into a similar error with `/tmp` nod existing on version 0.4.2 | darwin | aarch64 trying to run against an EKS provider.

Do you know when the next tag/bundled version is likely to happen? I'm trying to make a reproducible setup for my team and would rather not add the from-source installation steps to it if I don't have to.

**Comment by yufio** [comment]
@sanguivore-easyco unfortunately it still breaks for me on Version 0.5.0-11 (20231212.083322). I am now waiting for stable 0.5.0 maybe that solves it.

The error trying to rebuild my pod:

`[10:27:41] fatal Process exited with status 1
run agent command
github.com/loft-sh/devpod/pkg/devcontainer/sshtunnel.ExecuteCommand.func2
        /Users/runner/work/devpod/devpod/pkg/devcontainer/sshtunnel/sshtunnel.go:122
runtime.goexit
        /Users/runner/hostedtoolcache/go/1.20.5/x64/src/runtime/asm_amd64.s:1598`

**Comment by jmdacruz** [comment]
The fix referenced in #807 seems to be creating a `/tmp` folder, but I wonder if in the case of a Kubernetes provider it would be better to mount an empty directory for `/tmp` instead. I'm going to try and do this with the pod template feature (if I can find the default template)
